src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,130 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
+@SuppressWarnings("unchecked")
 public class ArrayList<T> implements List<T> {
+    public static final int INITIAL_CAPACITY = 10;
+    public static final String NO_SUCH_ELEMENT_MESSAGE = "There is no such element in the list - ";
+    public static final String OUT_OF_BOUNDS_MESSAGE = "Index out of bounds";
+
+    private Object[] storage;
+    private int size = 0;
+
+    public ArrayList() {
+        storage = new Object[INITIAL_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        ensureCapacity();
+        storage[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index - 1 == size) {
+            add(value);
+        } else {
+            ensureCapacity();
+            ensureAddition(index);
+            Object[] temp = new Object[storage.length];
+            System.arraycopy(storage, 0, temp, 0, index);
```

Comment:
we don't need this lines

Reply:
previous comment wasn't applied
we can remove 32 and 33 lines and use `storage` instead of `temp`